File: TransCoder_tokenized_test_set_functions/java/COUNT_SUBSTRINGS_WITH_SAME_FIRST_AND_LAST_CHARACTERS.java
static int countSubstringWithEqualEnds ( String s ) {
  int result = 0;
  int n = s . length ( );
  for ( int i = 0;
  i < n;
  i ++ ) for ( int j = i;
  j < n;
  j ++ ) if ( s . charAt ( i ) == s . charAt ( j ) ) result ++;
  return result;
}

File: results/py/COUNT_SUBSTRINGS_WITH_SAME_FIRST_AND_LAST_CHARACTERS.cpp
int f_filled ( string s ) {
  int result = 0 ;
  ;
  int n = s . length ( ) ;
  ;
  for ( int i = 0 ;
  i < n ;
  i ++ ) {
    for ( int j = i ;
    j < n ;
    j ++ ) {
      if ( ( s [ i ] == s [ j ] ) || ( s [ i ] == s [ j ] ) ) {
        result = result + 1 ;
      }
    }
  }
  return result ;
}
